cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/02/performance_testing.spec.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  DATASOURCE_NAME,
+  INDEX_PATTERN_WITH_TIME,
+  INDEX_WITH_TIME_1,
+  INDEX_WITH_TIME_2,
+} from '../../../../../../utils/constants';
+import {
+  generateAllTestConfigurations,
+  getRandomizedWorkspaceName,
+  setDatePickerDatesAndSearchIfRelevant,
+} from '../../../../../../utils/apps/query_enhancements/shared';
+import { generateDisplayTestConfiguration } from '../../../../../../utils/apps/query_enhancements/language_specific_display';
+import { prepareTestSuite } from '../../../../../../utils/helpers';
+
+const workspaceName = getRandomizedWorkspaceName();
+
+const runSavedSearchTests = () => {
+  describe('Performance testing', () => {
+    before(() => {
+      cy.osd.setupWorkspaceAndDataSourceWithIndices(workspaceName, [
+        INDEX_WITH_TIME_1,
+        INDEX_WITH_TIME_2,
+      ]);
+      cy.createWorkspaceIndexPatterns({
+        workspaceName: workspaceName,
+        indexPattern: INDEX_PATTERN_WITH_TIME.replace('*', ''),
+        timefieldName: 'timestamp',
+        dataSource: DATASOURCE_NAME,
+        isEnhancement: true,
+      });
+      cy.osd.navigateToWorkSpaceSpecificPage({
+        workspaceName,
+        page: 'discover',
+        isEnhancement: true,
+      });
+    });
+
+    after(() => {
+      cy.osd.cleanupWorkspaceAndDataSourceAndIndices(workspaceName, [
+        INDEX_WITH_TIME_1,
+        INDEX_WITH_TIME_2,
+      ]);
+    });
+
+    generateAllTestConfigurations(generateDisplayTestConfiguration).forEach((config) => {
+      const keywords = ['PPL', 'DQL', 'SQL', 'Lucene'];
+      const matchingKeyword = keywords.find((keyword) => config.language.includes(keyword));
```

Comment:
if you create your own `generatePerformanceTestConfiguration`, you can just have this as a prop within config

Reply:
better yet maybe `eventName` itself should be a config